lib/common/httpclient.c

```diff
@@ -302,3 +302,9 @@ void h2o_httpclient_set_conn_properties_of_socket(h2o_socket_t *sock, h2o_httpcl
     properties->ssl.cipher_bits = h2o_socket_get_ssl_cipher_bits(sock);
     properties->sock = sock;
 }
+
+static int socket_stealed(h2o_httpclient_t *client, const char *errstr, h2o_header_t *trailers, size_t num_trailers)
+{
+    assert(!"this callback is solery for signaling purpose and should never be called");
+}
+const h2o_httpclient_body_cb h2o_httpclient_socket_stealed = socket_stealed;
```

Comment:
Past participle of "steal" is "stolen", though I think we can call the function "h2o_httpclient_steal_socket".

Also, there is no need to define a static function and to have a global that takes the address. All we need is an external function with the correct name, e.g.,:
```suggestion
int h2o_httpclient_steal_socket(h2o_httpclient_t *client, const char *errstr, h2o_header_t *trailers, size_t num_trailers)
{
    assert(!"this callback is solely for signaling purpose and should never be called");
}
```

Reply:
Renamed the function in https://github.com/h2o/h2o/pull/3405/commits/d59ab4ab39d73ebe37956a0115477d54d097cf9d.
OTOH I'd think that the current way of using `extern const` prevents from repeating body_cb's arguments in httpclient.h, and it looks like a little clearer that it's just a constant to signal something, not the actual routine which would be invoked. What do you think?

Reply:
A function is a constant pointer (in such that you cannot alter the address or the contents).

Reply:
sorry what do you mean?

Reply:
Regarding your argument that it is good to be explicit that something is a constant pointer, my point is that a function declaration is exactly about declaring a constant pointer.